domain/secret/modelmigration/export.go

```diff
@@ -47,11 +47,10 @@ func (e *exportOperation) Name() string {
 }
 
 func (e *exportOperation) Setup(scope modelmigration.Scope) error {
-	// We must not use a watcher during migration, so it's safe to pass a
-	// nil watcher factory.
 	e.service = service.NewSecretService(
 		state.NewState(scope.ModelDB(), e.logger),
 		secretbackendstate.NewState(scope.ControllerDB(), e.logger),
+		nil,
```

Comment:
I'm starting to think there should be a `service.NewImportSecretService` so you don't have to do this nonsense.

Reply:
I left this as the last commit in the list, so I could potentially revisit and amend. Having done the demonstration, I may leave this to @wallyworld for when he applies this technique to the backend service.

Reply:
Also, when importing you wouldn't even care about leadership/lease (we guarantee it's the same after import), so it would simplify the importing logic.